use different registration confirmation url if config'd

bdewilde opened the pull request:
## changes

- adds `FE_APP_SITE` to app config, for use in production, so that confirmation URLs link to the front-end app rather than the back-end api, where we can handle them cleanly
- builds a different confirmation url for registration confirmation and password reset if this configuration is specified for the app

## context

https://app.asana.com/1/6325821815997/project/1206730431337718/task/1209811648565213?focus=true

## questions

- is this what you had in mind?
- does this _work?_
- should i do something similar for other confirmation links?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1210036871321400

bdewilde committed:
add optional server name config

bdewilde committed:
customize register confirm url

sophiaharms commented:
I'm not able to build from this branch. When I do, http://localhost:5001/docs returns:

{
  "error": "Internal Server Error",
  "message": "404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again."
}

bdewilde commented:
@sophiaharms sure enough, that localhost url doesn't work when `COLANDR_SERVER_NAME` is included in `.env`. however, this config variable is intended for production use, not dev -- does it work _in production?_ or is it having unwanted side-effects in flask's functionality?

i could always follow the example of what I did for v1.0, where a custom server name is included (optionally) as a parameter to the endpoint that generates the confirmation url. what do you think?

sophiaharms commented:
@bdewilde I'm not able to get this working in production either. A gunicorn error causes colandr-api to stop. It's spawning 16+ workers, which it does not do on develop. I think I'm just running out of disk space, but I'm running out because something's going wrong here

bdewilde committed:
Merge branch 'develop' into change-register-confirm-url

bdewilde committed:
style: move config param, add newline

bdewilde committed:
rename external email envvar / app cfg key

SERVER_NAME is a weird and problematic configuration key in Flask, and we should avoid it

bdewilde committed:
use new app cfg in register confirm email url

bdewilde committed:
use new app cfg in passwd reset confirm email url

bdewilde committed:
fix: avoid double slash in confirm urls
